src/dataframe/dataframe.jl

```diff
@@ -952,23 +952,6 @@ Base.convert(::Type{DataFrame}, d::AbstractDict) = DataFrame(d)
 ##
 ##############################################################################
 
-function Base.push!(df::DataFrame, associative::AbstractDict{Symbol,Any})
```

Comment:
AFAICT this method was provided separately because it only works with `Symbol` keys, while the one below only works with `AbstractString` keys. The latter is unusual since I think we generally require symbols, so it could be deprecated (and its `get` call is really weird).

Reply:
OK. But - as noted above - this `get` has the same performance as `getindex` and `push!` for `:Symbol` was implemented in a bad way as it required `Any` value. I will revert and fix it and deprecate the other one.

Reply:
I am removing the tests for deprecated version of the method.

Reply:
Actually I had to use other approach for deprecation as we can have `Dict{Any}` or `Dict{Union{Symbol, Missing}}` that actually contains only `Symbol` as keys.